pygmt/src/histogram.py

```diff
@@ -0,0 +1,60 @@
+"""
+Histogram - Create a histogram
+"""
+from pygmt.clib import Session
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    A="horizontal",
+    B="frame",
+    C="cmap",
+    G="fill",
+    J="projection",
+    R="region",
+    T="series",
```

Comment:
Is `series` a good name? Other Python histogram functions (e.g., [`numpy.histogram`](https://numpy.org/doc/stable/reference/generated/numpy.histogram.html), [`plt.histogram`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.hist.html) use `bins` instead.

Reply:
See my comment on https://github.com/GenericMappingTools/pygmt/pull/1072#discussion_r611029660. Probably not a good idea to use `bins` as the input for `-T` is structured quite differently to the `bins` argument in `np.histogram` and `plt.histogram`.

Reply:
Actually, it depends on how we want to implement the **-T** option in PyGMT.

histogram **-T** have following usages:

1. -T6
2. -T0/200/5
3. -Tfile
4. -T0,3,4,6,10

The current implementation doesn't work well for usage 4, since users have to pass a string "0,3,4,6,10". 

Instead, if we adopt the syntax of `np.histogram`, we can use more Pythonic syntax
```
bins=6
bins=np.arange(0, 200, 5)
bins="file"
bins=[0, 3, 4, 6, 10)
```

Reply:
Currently we're using `@kwargs_to_strings(T="sequence")` which implies Option 2 (-T0/200/5) and I think this works well for `makecpt`/`grd2cpt` where regular sized bins are the norm, and also `histogram` to some extent. Doing Option 4 would mean using `@kwargs_to_strings(T="sequence_comma")` and I guess this would be fine for people wanting irregular sized bins for `histogram`.

I don't often plot histograms, so I'll leave it up to you all to decide which one is better. Ideally we would standardize the `-T` option across all modules, but we could also use `series` (with T="sequence") ` for `makecpt`/`grd2cpt`, and `bins` (with T="sequence_comma")  for `histogram` if that makes more sense for new users.

Reply:
Just a note here: we don't have to use `@use_alias` and `@kwargs_to_strings` for **-T**.

Instead, we can check if the `bins` (or any other parameter name) argument is a string, one value, three values, or a list of values. I think it provides the most Pythonic syntax and also the greatest flexibility.

Reply:
I'm not looking to shirk responsibility for wrapping `histogram`, but are many geoscientists creating histograms with GMT? I think it's good to provide the capability in PyGMT in case anyone is looking to incorporate it into their figures, but I imagine almost all PyGMT users will be familiar with numpy and matplotlib and using that instead for plots like this, so I don't wish to make the PyGMT syntax overly different than GMT syntax.

Reply:
Good point. For a single histogram plot, it would be be much easier to use something like the built-in [pandas hist function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.hist.html) (e.g. just do `df.hist()`). But for say, mixing a `grdview` plot with a `histogram` as an 'inset', then using PyGMT would be better :wink:

Let's get another opinion. Ping @core-man since he seemed interested in `histogram` at https://github.com/GenericMappingTools/pygmt/issues/563#issuecomment-801246659. What syntax do you think PyGMT's `histogram` should follow, GMT's or Matplotlib's?

Reply:
During my 10-year seismological studies, I only use GMT to plot figures and sometimes use AI/3D max to plot cartoons. I began to use Python and PyGMT at the same time last year, so I have never learned matplotlib. So for my study, GMT/PyGMT is enough right now. For the PyGMT scope I understand now, we'd like to provide a good Python interface for GMT. So as we go to v1.0.0, I prefer more Pythonic syntax instead of strict GMT's. At that time, I think (also hope) most GMT users will learn GMT by using PyGMT.

For this issue, I think the current implementation is okay and most users can use it well (I never use option 4). However, it's better we could refractor all the similar parameters to be more Pythonic syntax as we go to v1.0.0, e.g.,
```
bins=6
bins=np.arange(0, 200, 5)
bins="file"
bins=[0, 3, 4, 6, 10)
```

Reply:
I agree that we should move to a more Pythonic syntax overall. Assuming that is the eventual plan, it's creating more work for our future selves by wrapping new features with a GMT syntax. That being said, I think there are more commonly-used features (setting map projections, updating pen/fill settings, adding a frame) that should be prioritized if the plan is to begin incorporating Pythonic syntax. This is certainly an issue outside of the scope of wrapping `histogram`, but I think if we are changing new features to be in-line with a more Pythonic syntax we should be also updating our old features ASAP.

Reply:
Maybe I am missing it, but is there a consensus on how this function should be modified?

Reply:
For the sake of time and consistency with other PyGMT modules like makecpt/grd2cpt, I say we stick with using `series` (-T) instead of `bins`. For better or worse, we are still heavily skewed towards the style of GMT core rather than NumPy, and decisions of what is 'Pythonic' should go in a separate issue. It is better to be consistently like GMT, rather than confuse users with a mix of NumPy-like and GMT command-line-like syntax. So @willschlitzer, let's keep things as it is for now :)